### PR TITLE
WIP to add support for Geometry (POINT) type

### PR DESCRIPTION
**Pull request by siemensikkema**
Points can now be correctly stored and retrieved.

- [x] Geometry SQL Data type
- [x] Store point values
- [x] API docs
- [x] Tests
- [ ] Support lineString (well known binary type 2)
- [ ] Support polygon (3)
- [ ] Support multiPoint (4)
- [ ] Support multiLineString (5)
- [ ] Support multiPolygon (6)
- [ ] Support geometryCollection (7)

### CODE REVIEW

File: Sources/MySQL/Connection/MySQLGeometry.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            return "Point(\(x) \(y))"
+        }
+    }
+}
+
+extension MySQLGeometry {
+    func convertToData() -> Data {
+        switch self {
+        case .point(let x, let y):
+            let bufferSize = MySQLGeometry.headerSize + 2 * MemoryLayout<Double>.size
```
Comment (BrettRToomey) [inline Sources/MySQL/Connection/MySQLGeometry.swift]: The MySQL spec requires these to be [8 bytes](https://dev.mysql.com/doc/refman/5.7/en/gis-data-formats.html#gis-wkb-format). I don't know about in Swift, but in C and C++ a double isn't guaranteed to be 8 bytes. Just to have higher precision than a float.

> WKB uses 1-byte unsigned integers, 4-byte unsigned integers, and 8-byte double-precision numbers (IEEE 754 format). A byte is eight bits.

File: Sources/MySQL/Connection/MySQLGeometry.swift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        guard count >= headerSize else {
+            throw MySQLError(identifier: "geometryType", reason: "Not enough data to read geometry type.")
+        }
+        var buffer = ByteBufferAllocator().buffer(capacity: count)
+        buffer.withUnsafeMutableWritableBytes { (p) in
+            p.copyBytes(from: data)
+        }
+
+        let _: UInt32 = try buffer.requireInteger()
+        let _: UInt8 = try buffer.requireInteger()
```
Comment (BrettRToomey) [inline Sources/MySQL/Connection/MySQLGeometry.swift]: I don't know if it's possible, but you may want to assert that this is little-endian.

File: Sources/MySQL/Connection/MySQLGeometry.swift [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            )
+        default:
+            throw MySQLError(identifier: "geometryType", reason: "Only 'Point' geometry data type is supported.")
+        }
+    }
+}
+
+extension MySQLGeometry: ReflectionDecodable {
+    public static func reflectDecoded() throws -> (MySQLGeometry, MySQLGeometry) {
+        return (.point(x: 0, y: 0), .point(x: 0, y: 1))
```
Comment (BrettRToomey) [inline Sources/MySQL/Connection/MySQLGeometry.swift]: What's this?

File: Sources/MySQL/Connection/MySQLGeometry.swift (hunk 1 above)
Comment (BrettRToomey) [reply Sources/MySQL/Connection/MySQLGeometry.swift]: In C we'd typically use a static assert to ensure that the double is the size we're expecting or update the algorithm to allow for a double to be another size

File: Sources/MySQL/Connection/MySQLGeometry.swift (hunk 3 above)
Comment (siemensikkema) [reply Sources/MySQL/Connection/MySQLGeometry.swift]: Ah, the wonderful world of `ReflectionDecodable`, see https://github.com/vapor/core/blob/master/Sources/Core/CodableReflection/ReflectionDecodable.swift. However, it seems I don't actually need this one anymore. I believe I needed it earlier to exclude a property on my model with this type, the process of which required reflection.

File: Sources/MySQL/Connection/MySQLGeometry.swift (hunk 2 above)
Comment (siemensikkema) [reply Sources/MySQL/Connection/MySQLGeometry.swift]: Will do

**Comment by siemensikkema** [comment]
I  cannot work on this before august 6 the earliest so if anyone else has time to pick up the unit tests, that would be awesome!

**Comment by tanner0101** [comment]
Moving to https://github.com/vapor/mysql-nio/issues/42

**Pull request closed without merging**